snapcraft/plugins/nodejs.py

```diff
@@ -178,19 +178,27 @@ def build(self):
                 self._manifest["yarn-lock-contents"] = lock_file.read()
 
         # Get the names and versions of installed packages
-        installed_node_packages = self._get_installed_node_packages(self.installdir)
-        self._manifest["node-packages"] = [
-            "{}={}".format(name, installed_node_packages[name])
-            for name in installed_node_packages
-        ]
+        if self.options.nodejs_package_manager == "npm":
+            installed_node_packages = self._get_installed_node_packages(self.installdir)
+            self._manifest["node-packages"] = [
+                "{}={}".format(name, installed_node_packages[name])
+                for name in installed_node_packages
+            ]
+        # Skip this step if yarn is used, as it may produce different
+        # dependency trees than npm
+        else:
+            self._manifest["node-packages"] = []
 
-    def _install(self, rootdir):
+    def _install_node_and_yarn(self, rootdir):
         self._nodejs_tar.provision(self._npm_dir, clean_target=False, keep_tarball=True)
         if self.options.nodejs_package_manager == "yarn":
             self._yarn_tar.provision(
                 self._npm_dir, clean_target=False, keep_tarball=True
             )
+        # Check to see if package.json exists in pull step
```

Comment:
Curious, what's the point of this?

Reply:
I wanted it to crash during the pull step if it couldn't find the package.json, since that's what would happen previously, but it's not strictly necessary.

Reply:
Great, thanks - makes sense!

To clarify, by "crash", you are referring to the error message "Could not find a 'package.json' in the source tree...", correct?

Reply:
Yes, that's right.